Cargo.lock

```diff
@@ -184,6 +184,12 @@ version = "0.5.6"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "0e4cec68f03f32e44924783795810fa50a7035d8c8ebe78580ad7e6c703fba38"
 
+[[package]]
+name = "bytes"
```

Comment:
The double 'bytes' dependency is unfortunate; if we do take this we'll probably want to see what depends on 'bytes' 0.5.6.

Reply:
I'm not very experienced with the Rust build infrastructure, what should I do to resolve this?

Reply:
`cargo tree --invert bytes` can show all the packages that depend on `bytes`; fortunately…

```
bytes v0.5.6
├── libsignal-protocol v0.1.0 (/Users/jordan/dev/libsignal-client/rust/protocol)
├── prost v0.6.1
│   ├── libsignal-protocol v0.1.0 (/Users/jordan/dev/libsignal-client/rust/protocol)
│   ├── prost-build v0.6.1
│   │   [build-dependencies]
│   │   └── libsignal-protocol v0.1.0 (/Users/jordan/dev/libsignal-client/rust/protocol)
│   └── prost-types v0.6.1
│       └── prost-build v0.6.1 (*)
├── prost-build v0.6.1 (*)
└── prost-types v0.6.1 (*)
```

…the only other user seems to be libsignal-protocol itself! So you can bump that version as well in the Cargo.toml.

Reply:
Looks like `cargo tree -d -e normal --workspace` shows that `bytes` is also an indirect dependency for `jni` through `combine` and it's using a much older version. Is the only solution to update it in `combine` upstream?

```
bytes v0.5.6
└── combine v4.3.2
    └── jni v0.17.0
        ├── libsignal-bridge v0.1.0 (/home/siva/src/libsignal-client/rust/bridge/shared)
        │   ├── libsignal-ffi v0.1.2 (/home/siva/src/libsignal-client/rust/bridge/ffi)
        │   ├── libsignal-jni v0.2.3 (/home/siva/src/libsignal-client/rust/bridge/jni)
        │   └── libsignal-node v0.3.0 (/home/siva/src/libsignal-client/rust/bridge/node)
        └── libsignal-jni v0.2.3 (/home/siva/src/libsignal-client/rust/bridge/jni)

bytes v1.0.1
├── libsignal-protocol v0.1.0 (/home/siva/src/libsignal-client/rust/protocol)
│   ├── libsignal-bridge v0.1.0 (/home/siva/src/libsignal-client/rust/bridge/shared) (*)
│   ├── libsignal-ffi v0.1.2 (/home/siva/src/libsignal-client/rust/bridge/ffi)
│   ├── libsignal-jni v0.2.3 (/home/siva/src/libsignal-client/rust/bridge/jni)
│   └── libsignal-node v0.3.0 (/home/siva/src/libsignal-client/rust/bridge/node)
└── prost v0.7.0
    └── libsignal-protocol v0.1.0 (/home/siva/src/libsignal-client/rust/protocol) (*)
```

Reply:
Oops, I always forget `--workspace` when I run it manually. It looks like the 4.5.x series of `combine` supports bytes v1, but there's not a new enough `jni` to use that `combine`. So it looks like we have a trade-off here: add duplicate code size to our resulting binary, or don't support Alpine Linux. I'm trying to think of ways to support both but have the right defaults, but they'd all mean threading through Cargo features all over the place.